crates/configuration/src/version4/mod.rs

```diff
@@ -188,6 +200,85 @@ pub async fn introspect(
     })
 }
 
+/// given scalar type names already in use, introspected tables, and optionally any existing table configuration:
+/// get collections with names guaranteed unique, preserving customized collection and field names if any
+fn get_aliased_tables(
+    type_names: HashSet<TypeName>,
+    tables: metadata::TablesInfo,
+    old_tables: &metadata::TablesInfo,
+) -> metadata::TablesInfo {
+    let mut type_names = type_names;
+    let mut mapped_tables = BTreeMap::new();
+
+    for (collection_name, table_info) in tables.0 {
```

Comment:
I think I remember you already verified that this will iterate names in a consistent order so that's good